lib/active_triples/term.rb

```diff
@@ -26,12 +26,13 @@ def clear
       set(nil)
     end
 
-    def result
-      parent.query(:subject => rdf_subject, :predicate => predicate)
-        .each_with_object([]) do |x, collector|
-          converted_object = convert_object(x.object)
-          collector << converted_object unless converted_object.nil?
-        end
+    def result(convert=true)
+      results = parent.query(:subject => rdf_subject, :predicate => predicate)
+      convert ? convert(results) : results.to_a
+    end
+
+    def convert(results)
```

Comment:
`#convert_object` is protected.  Is there a reason we want to expose it if you pass an Enumerable?